lib/driver.rb

```diff
@@ -0,0 +1,83 @@
+require 'csv'
+require 'time'
+require_relative 'csv_record'
+require_relative 'trip_dispatcher'
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :id, :name, :vin, :trips
+    attr_accessor :status
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE, trips: nil)
+      super(id)
+      @name = name
+      @vin = vin #17 char
+      @status = status
+      @trips = trips || []
+      
+  
+      if status.class == String
+        @status = status.to_sym
+      end
+
+      if vin.class == Integer
+        vin.to_s
+      end
+
+      valid_statuses = [:AVAILABLE, :UNAVAILABLE]
+      raise ArgumentError if !valid_statuses.include?(@status)
+
+      raise ArgumentError if id <= 0 || id.nil?
+
+      raise ArgumentError if vin.length != 17
+    end 
+
+    def add_trip(trip)
+      trips << trip
+    end
+    
+    def average_rating
+      total_ratings = 0
+      @trips.each do |trip|
+        total_ratings += trip.rating
+      end
```

Comment:
You can simplify this using `sum`:

```suggestion
      total_ratings = @trips.sum { |trip| trip.rating }
```